src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +471,33 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        ostype = os.name
+        temp_stage = self._session.get_session_stage()
+        if os.path.exists(path) and os.path.isfile(path):
+            _ = self._session.file.put(path, temp_stage, auto_compress=False)
+            _, filename = os.path.split(path)
+            path = os.path.join(temp_stage, filename)
```

Comment:
We should probably create a `stage_path` variable for this. Similar to directory one

Reply:
OK, I will create a stage_path variable at the beginning of the fucntion